probe-rs/src/memory/mod.rs

```diff
@@ -277,19 +277,23 @@ where
         assert!(end_extra_count < 4);
         assert!(inbetween_count % 4 == 0);
 
-        // If we do not have 32 bit aligned access we first check that we can do 8 bit aligned access on this platform.
-        // If we cannot we throw an error.
-        // If we can we read the first n < 4 bytes up until the word aligned address that comes next.
-        if address % 4 != 0 || len % 4 != 0 {
-            // If we do not support 8 bit transfers we have to bail because we can only do 32 bit word aligned transers.
+        if start_extra_count != 0 || end_extra_count != 0 {
+            // If we do not support 8 bit transfers we have to bail
+            // because we have to do unaligned writes but can only do
+            // 32 bit word aligned transers.
             if !self.supports_8bit_transfers()? {
                 return Err(MemoryNotAlignedError {
                     address,
                     alignment: 4,
                 }
                 .into());
             }
+        }
 
+        // If we do not have 32 bit aligned access we first check that we can do 8 bit aligned access on this platform.
+        // If we cannot we throw an error.
+        // If we can we read the first n < 4 bytes up until the word aligned address that comes next.
```

Comment:
This comment doesn't fit here. (The check already happened above.)

Reply:
Good shout, we also don't attempt to do an aligned read (yet? anymore?)